src/components/Workspace.jsx

```diff
@@ -133,9 +138,88 @@ export default class Workspace extends React.Component {
     );
   }
 
+  _renderHiddenLanguages() {
+    const {
+      currentProject,
+      hiddenLanguages,
+      onComponentToggle,
+    } = this.props;
+    return (
+      <React.Fragment>
+        {hiddenLanguages.map(({language}) =>
+          (
+            <CollapsedComponent
+              component={`editor.${language}`}
+              key={language}
+              projectKey={currentProject.projectKey}
+              text={t(`languages.${language}`)}
+              onComponentUnhide={onComponentToggle}
+            />
+          ))}
+      </React.Fragment>
+    );
+  }
+
+  _renderHiddenLeftColumnComponents() {
+    const {
+      hiddenLanguages,
+    } = this.props;
+
+    if (
+      hiddenLanguages.length === 0 ||
+      hiddenLanguages.length === LANGUAGES.length
+    ) {
+      return null;
+    }
+
+    return this._renderHiddenLanguages();
+  }
+
+  _renderHiddenRightColumnComponents() {
+    const {
+      currentProject,
+      onComponentToggle,
+    } = this.props;
+    const rightColumnComponents = ['console'];
+    return rightColumnComponents.
```

Comment:
So unless I’m misreading here, `rightColumnComponents` is always a one-element array containing just `'console'`, right? In which case it seems like the `filter`/`map` of it all is a bit of overkill?

Reply:
For now...the thought was the next revision is going to add the preview to the list of right column components and then it'll just be adding that to the array.